source/common/config/well_known_names.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <string>
+
+#include "common/common/singleton.h"
+
+namespace Envoy {
+namespace Config {
+
+/**
+ * Well-known network filter names.
+ */
+class NetworkFilterNameValues {
+public:
+  // Client ssl auth filter
+  const std::string CLIENT_SSL_AUTH = "client_ssl_auth";
```

Comment:
I'm wondering for the network/HTTP names whether we want to make it that we retain existing names without a `envoy.` prefix for v1, but for v2 make these `envoy.ratelimit` etc. We could just do the prefix step during the JSON -> proto translation steps and then we have a consistent naming scheme across all filters/pluggables for v2.